src/api/app/helpers/webui/comments_helper.rb

```diff
@@ -1,14 +1,20 @@
 module Webui::CommentsHelper
   def comment_user_role_titles(comment)
-    roles = comment.user.roles.global.pluck(:title)
-
-    roles += roles_for_project(comment) if comment.commentable.is_a?(Project)
-
-    roles += roles_for_package(comment) if comment.commentable.is_a?(Package)
+    roles = []
 
-    roles += roles_for_request(comment) if comment.commentable.is_a?(BsRequest)
+    # binding.pry
+    case
+    when comment.commentable.is_a?(BsRequest)
+      roles = roles_for_request(comment)
+    when comment.commentable.is_a?(Project)
+      # roles.push('maintainer') if comment.user.can_modify?(comment.commentable)
+      roles.push('maintainer') if roles_for_project(comment).any?
```

Comment:
Is there any reason why you don't use `roles << 'maintainer'`?

Reply:
I personally prefer the `push` format, it makes it visually more clear to me what is on the left and on the right of the `if` due to the `.push(...` concatenation:

`roles.push('maintainer') if roles_for_project(comment).any?`
VS
`roles << 'maintainer' if roles_for_project(comment).any?`

Is there any style preference for that or any other reason I am missing to prefer one or the other?

Reply:
I was missing the _speed_ diff, thanks for pointing it out. I'm adapting to that then :+1:

Reply:
Mmmm..do you have any pointers to read about the performance difference? I've been searching for it and I can only find the fact that `push` accept multiple and `<<` doesn't, and `<<` can have lower or higher precedence because it's an operator, `push` it's more _linear_ instead.

Ruby's doc doesn't mention anything about it [1]

[1] https://ruby-doc.org/3.2.2/Array.html#class-Array-label-Adding+Items+to+Arrays

Reply:
No, I don't. `<<` is an operator. `push` is a method. I had the impression that operators are faster than methods, but I could not find anything to back this up. So then my reasoning falls back to `<<` is the convention used everywhere when concatenating only one object.